src/foolscap/banana.py

```diff
@@ -346,8 +346,11 @@ def pushSlicer(self, slicer, obj):
         # methods which are *not* generators.
 
         itr = slicer.slice(topSlicer.streamable, self)
-        next = iter(itr).next
-
+        if six.PY2:
+            next = iter(itr).next
+        else:
+            next = iter(itr).__next__
+            
         # we are now committed to sending the OPEN token, meaning that
```

Comment:
same (don't shadow a global). Also there was @habnabit 's suggestion of `nextfn = partial(...)` from the other (?) PR.

Reply:
Fixed to avoid global shadowing, but using functools.partial is an overkill here.